draft-ietf-quic-tls.md

```diff
@@ -919,6 +930,80 @@ the connection that is hosted on stream 0.  This sequence number is not visible
 to QUIC.
 
 
+## Packet Number Protection {#pn-encrypt}
+
+QUIC packets are protected using a key that is derived from the current set of
+secrets.  The key derived using the "pn" label is used to protect the packet
+number from casual observation.  The packet number protection algorithm depends
+on the negotiated AEAD.
+
+Packet number protection is applied after packet protection is applied (see
+{{aead}}).  The ciphertext of the packet is sampled and used as input to an
+encryption algorithm.
+
+For packets with a long header, the ciphertext starting immediately after the
+packet number is used.
+
+For packets with a short header, the packet number length is not known before
+decryption, so it is assumed to be the smaller of the maximum possible packet
```

Comment:
I'm a bit confused. Why is the packet number length not known before hand? Doesn't the short packet header still indicate the length?

Reply:
This is in preparation for going to varint packet numbers.  It would need a complete rewrite if this assumption were lifted.

Reply:
But this PR doesn't change the short header to use varint packet numbers... unless I missed something. Assuming I didn't, it looks like one of three things need to happen then:

1. An additional PR that changes short headers to varint packet numbers needs to be merged first, and then this rebased on that.
2. This PR should also change the short headers to varint packet numbers.
3. This PR should be written to the current format of short header packet numbers.

If we are going to do it, I don't see why we don't just do (2) right now. Do you expect that to be a huge change?

Reply:
Given where this PR is right now, given that the chairs have green-lighted this, and given that we're unicorning, I recommend landing this PR as is. Nick, can you open an issue to track this editorial issue, so that it doesn't get dropped?

Reply:
I can open an issue, no problem. But we have waited and discussed all this time to ensure we get a good solution. Can't we take a bit longer just to iron out the details and get a complete solution merged all at once? It should be trivial enough to change the short header packet number encoding to varints.

Reply:
I have fixed the incorrect statement, but left the design unchanged.